src/base/IoManip.h

```diff
@@ -187,32 +187,57 @@ class AsList
     /// a c-string to print between consecutive items (if any). Caller must ensure lifetime.
     auto &delimitedBy(const char * const d) { delimiter = d; return *this; }
 
+    /// c-string to print before and after each item. Caller must ensure lifetime.
+    auto &quoted(const char * const q = "\"") { preQuote = postQuote = q; return *this; }
+
+    /// c-strings to print before and after each item. Caller must ensure lifetime.
+    auto &quoted(const char * const preQ, const char * const postQ) { preQuote = preQ; postQuote = postQ; return *this; }
```

Comment:
I am not sure we should add this method (now). We do not need it right away, and it may trigger somewhat-difficult-to-spot bugs like this:

```C++
os << distraction()  << AsList(anotherDistraction()).quoted("(", "}") << moreDistractions(SBuf{});
```

Here is a different useful method that will cover most of the same use cases:

```C++
/// A character to print before each item. A matching character will be printed after each item.
auto &bracketed(const char openingCh)
{
    preQuote = openingBracket(openingCh);
    postQ = closingBracket(openingCh);
    return *this;
}
```

... which will result in a safer, simpler caller:

```C++
os << distraction()  << AsList(anotherDistraction()).bracketed('(') << moreDistractions(SBuf{});
```

Unfortunately, it may not be possible (without heroic efforts) to implement bracketed() so that it asserts at compile time if the wrong character is used as a parameter (e.g., `}` instead of `{`). I have not tried.

In defense of the current quoted(pre, post) method, most of the bugs like the one sketched above will be caught by good editors/IDEs while unbalanced brackets introduced by bracketed() will confuse some editors/IDEs.

The two methods in questions are not mutually exclusive, of course. However, IMO, we should not add methods that will not receive obvious, immediate use: Unused code is a red flag, and some method details may actually be influenced by use cases.

I do not insist on this method removal, but, if you agree that is should not be added (now), applying this suggestion will remove it cleanly AFAICT:

```suggestion
```

Reply:
Bracketed() feels overengineered for the current use case.
Even quoted with two arguments suffers to a lesser extent the same problem, as you are rightly observing.